EIPS/eip-4834.md

```diff
@@ -0,0 +1,221 @@
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/erc-4834-hierarchical-domains-standard/8388
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-02-22
+requires: 165
+---
+
+## Abstract
+`ERC-4834` is a standard for generic name resolution with access control. It permits contracts that implement `ERC-4834` ("domains") to be addressable with a more human-friendly name, similar to `ERC-137`.
+
+Unlike `ERC-137`, any program that resolves domains should treat domains as equivalent to their resolved addresses. In practice, this means that instead of having to specify a contract address that looks like `0x6b175474e89094c44da98b954eedeac495271d0f`, it might be possible to put (this is a completely made-up address) `dai.token` and have it automatically resolve to the right address. This is different from `ERC-137`, where addresses are resolved by ENS to a "resolver," which then gets resolved to the correct address. 
+
+Another notable divergence from `ERC-137` is that access control can be extremely complex. `ERC-137` domains have a defined owner, whereas any access control patterns can be implemented for an `ERC-4834` domain. This allows for many more use-cases, such as a token that could be redeemed for domains.
+
+## Motivation
+The motivation behind this ERC is that existing standards like ERC-137 are relatively complex and have limitations. The advantage of ERC-4834 is that it provides a minimal interface that supports name resolution. This permits use-cases such as referencing an ERC-20 or ERC-1155 token by a name directly, without using a "resolver" contract.
+
+This ERC also implements flexible access control. This allows for clever implementations, such as having a token that can be redeemed for subdomains of the token, and domains that are themselves NFTs that grant the ability to manage the domain to the owner of said NFT. There can also be multiple managers of a single domain with different access, instead of a single owner. Of course, it's still possible to implement a single-owner domain, should that be desired.
+
+In addition, TLDs (like .com), subdomains, and even the "root" itself (analogous to the ENS registry) are all implemented as ERC-4834 smart contracts, meaning that name-resolution is a relatively simple iterative algorithm, and there is no limit to the amount of nesting that is possible (e.g. 0.1.2.3.4.5.6.7.8.9.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z could be valid if the root contains z, and z contains y, and etc).
+
+## Rationale
+As previously mentioned in the abstract and motivation, the design of this ERC is different than that of ERC-137. ERC-137 has a strong tiering - the registrars update the ENS registry, which points to resolvers. In ERC-4834, every domain is effectively a registry, resolver, and registrar all in one. This is specifically designed to be a simple addressing system.
```

Comment:
`Rationale` has to go after `Specification`.

Also, the `Rationale` section should describe the reasons behind particular choices in the specification (a dumb example would be to explain why `canDeleteDomain` and `canDeleteSubdomain` both exist.) Compare that to the `Motivation` section, which describes why this EIP as a whole should exist.